### PR TITLE
[12.0][IMP] account_asset_management: Add analytic tags and propagate

### PR DESCRIPTION
**Pull request by joao-p-marques**
Add analytic tags to `asset.profile` and `asset.asset` and propagate them between these models and to the created `account.move.line`
Add tests for propagation of analytic data from asset to account.move.line

@Tecnativa
TT28974

ping @pedrobaeza @victoralmau

### CODE REVIEW

File: account_asset_management/tests/account_asset_test_data.xml [hunk 1]
```diff
@@ -78,6 +78,8 @@
       <field name="purchase_value" eval="12000.0"/>
       <field name="salvage_value" eval="2000.0"/>
       <field name="profile_id" ref="account_asset_profile_car_5Y"/>
+      <field name="account_analytic_id" ref="analytic.analytic_administratif" />
+      <field name="analytic_tag_ids" search="[('id', 'in', [ref('analytic.tag_contract')])]" />
```
Comment (pedrobaeza) [inline account_asset_management/tests/account_asset_test_data.xml]:
```suggestion
      <field name="analytic_tag_ids" eval="[(4, ref('analytic.tag_contract')]" />
```

File: account_asset_management/tests/test_account_asset_management.py [hunk 2]
```diff
@@ -186,7 +186,20 @@ def test_01_nonprorata_basic(self):
...
+                len(
+                    move.line_ids.filtered(
+                        lambda line: line.analytic_account_id ==
+                        self.env.ref("analytic.analytic_administratif") and
+                        line.analytic_tag_ids ==
+                        self.env.ref("analytic.tag_contract")
+                    )
+                ),
+                1,
+            )
```
Comment (pedrobaeza) [inline account_asset_management/tests/test_account_asset_management.py]:
```suggestion
            self.assertTrue(
                move.line_ids.filtered(
                    lambda line: line.analytic_account_id ==
                    self.env.ref("analytic.analytic_administratif") and
                    line.analytic_tag_ids ==
                    self.env.ref("analytic.tag_contract")
                 )
            )
```

Comment (joao-p-marques) [reply account_asset_management/tests/test_account_asset_management.py]: Thanks for the suggestion @pedrobaeza 
However, this check with the evaluation of the `len()` also asserted that the analytic data was not being propagated to the lines that it shouldn't, but only to the ones of type "expense": https://github.com/OCA/account-financial-tools/pull/1149/files#diff-5318cd3560cdb2d23d2889fe39c5b71745c140810b20f64f0c9aa5b24b8b602bR200-R204
We will lose that with this change... Is that a problem?

File: account_asset_management/tests/account_asset_test_data.xml (hunk 1 above)
Comment (pedrobaeza) [reply account_asset_management/tests/account_asset_test_data.xml]: Sorry, missing ending `)`

File: account_asset_management/tests/test_account_asset_management.py (hunk 2 above)
Comment (pedrobaeza) [reply account_asset_management/tests/test_account_asset_management.py]: You are not testing that, but testing that there's only line with such account. If you want to test that, filter by account type, and then `assertEqual` its content.

Comment (joao-p-marques) [reply account_asset_management/tests/test_account_asset_management.py]: Ok, thanks! I made the change. I now test that the lines with analytic data exist (`assertTrue`) and that they match the appropriate lines, as you suggested (`assertEqual`)

File: account_asset_management/tests/test_account_asset_management.py [hunk 3]
```diff
@@ -186,7 +186,24 @@ def test_01_nonprorata_basic(self):
...
+            self.assertTrue(move_lines_w_analytic_data)
+            self.assertEquals(
+                move_lines_w_analytic_data,
+                move.line_ids.filtered(
+                    lambda line: line.account_id == self.env.ref("account.a_expense")
+                    # Filter for expense lines
+                    # (associated with the profile's depreciation account)
+                )
+            )
```
Comment (pedrobaeza) [inline account_asset_management/tests/test_account_asset_management.py]: Cleaner code IMO is:

```suggestion
            expense_line = move.line_ids.filtered(
                lambda line: line.account_id == self.env.ref("account.a_expense")
            )
            self.assertEqual(
                expense_line.analytic_account_id,
                self.env.ref("analytic.analytic_administratif"),
            )
            self.assertEqual(
                expense_line.analytic_tag_ids,
                self.env.ref("analytic.tag_contract")
            )
```

Comment (joao-p-marques) [reply account_asset_management/tests/test_account_asset_management.py]: Thanks :+1: 
Much better, indeed :smile:

**Review by pedrobaeza — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Now to v13

**Pull request merged**